File: frontend/src/services/userService.js
import { userApi } from '../api/userApi';

export const userService = {
  getAllUsers: async (pageNo, pageSize) => {
    return await userApi.getAllUsers(pageNo, pageSize);
  },

  getUserById: async (id) => {
    return await userApi.getUserById(id);
  },

  assignRole: async (userId, roleName) => {
    return await userApi.assignRole(userId, roleName);
  },

  deleteUser: async (id) => {
    return await userApi.deleteUser(id);
  },
};

File: backend/src/main/java/com/payroll/member6/service/UserService.java
package com.payroll.member6.service;

import com.payroll.member6.dto.UserDTO;
import com.payroll.response.PagedResponse;

/**
 * ============================================================================
 * Member 06: User Service Interface
 * ============================================================================
 * 
 * Why This File Exists:
 * --------------------
 * Service contract for User administration, retrieval, role assignment, and deletion.
 * 
 * @author Senior Java Software Architect
 * @version 1.0.0
 */
public interface UserService {

    UserDTO getUserById(Long id);

    UserDTO getUserByUsername(String username);

    PagedResponse<UserDTO> getAllUsers(int pageNo, int pageSize, String sortBy, String sortDir);

    UserDTO assignRoleToUser(Long userId, String roleName);

    void deleteUser(Long id);
}
